configs/tdb_external_kvstore_with_qspif.json

```diff
@@ -0,0 +1,123 @@
+                    "|         KVSTORE          |",
+                    "|                          |",
+                    "+--------------------------+ <-+ storage_tdb_internal.internal_base_address",
+                    "|                          |",
+                    "|        Bootloader        |",
+                    "|                          |",
+                    "+--------------------------+ <-+ 0",
+                    "Firmware Candidate Storage is on external QSPIF device at an offset of 0 MiB.",
+                    "The storage size is the size of the maximum application size (i.e. Active",
+                    "Application region size) multiplied by the number of storage-locations.",
+                    "(storage-locations is set to 1 in this configuration)",
+                    "+--------------------------+<-+ End of QSPIF block device",
+                    "|                          |",
+                    "+--------------------------+<-+ update-client.storage-size + update-client.storage-address",
+                    "|                          |",
+                    "+--------------------------+",
+                    "|                          |",
+                    "|   Firmware Candidate 0   |",
+                    "|                          |",
+                    "+--------------------------+",
+                    "|   Firmware Candidate 0   |",
+                    "|     Metadata Header      |",
+                    "+--------------------------+ <-+ update-client.storage-address (1024 * 1024)",
+                    "|                          |",
+                    "|     External KVStore     |",
+                    "|      Size: 1MB           |",
+                    "|                          |",
+                    "+--------------------------+ <-+ Start of QSPIF block device (i.e. 0x0); "
+            ]
+        },
+        "kvstore-size": {
+            "help": "The size of the area of flash reserved for kvstore, it neeeds to be even number of flash sectors",
+            "macro_name": "KVSTORE_SIZE",
+            "required": true
+        }
+    },
+    "macros": [
+        "ARM_UC_USE_PAL_BLOCKDEVICE=1",
+        "MBED_CLOUD_CLIENT_UPDATE_STORAGE=ARM_UCP_FLASHIAP_BLOCKDEVICE",
+        "MBED_BOOTLOADER_ACTIVE_HEADER_REGION_SIZE=1024"
+    ],
+    "target_overrides": {
+        "*": {
+            "target.console-uart"                       : false,
+            "target.c_lib"                              : "small",
+            "platform.use-mpu"                          : false,
+            "platform.default-serial-baud-rate"         : 115200,
+            "platform.stdio-flush-at-exit"              : false,
+            "update-client.firmware-header-version"     : "2",
+            "target.restrict_size"                      : "0x8000",
+            "kvstore-size"                              : "2*16*1024",
+            "mbed-bootloader.bootloader-size"           : "(64*1024)",
```

Comment:
this should be 32*1024, and then for F746NG specifically override to 64 KiB ?
Same thing for the restrict_size calculation.

Reply:
Most other boards will probably fit the bootloader to 32 Kib ?

Reply:
actually the restrict_size is already for 32 KiB size

Reply:
Fixed

Reply:
If restrict_size not defined, mbed compile uses 32KiB value